packages/workflow/test/ExpressionExtensions/ObjectExtensions.test.ts

```diff
@@ -27,68 +42,119 @@ describe('Data Transformation Functions', () => {
-				),
-			).toEqual({
-				test1: 'i exist',
+		describe('.keepFieldsContaining', () => {
+			test('.keepFieldsContaining should work on an object', () => {
+				expect(
+					evaluate(
+						'={{ ({ test1: "i exist", test2: "i should be removed", test3: "i should also be removed" }).keepFieldsContaining("exist") }}',
+					),
+				).toEqual({
+					test1: 'i exist',
+				});
 			});
-		});
 
-		test('.keepFieldsContaining should work on a nested object', () => {
-			expect(
-				evaluate(
-					'={{ ({ test1: "i exist", test2: "i should be removed", test3: { test4: "me too" } }).keepFieldsContaining("exist") }}',
-				),
-			).toEqual({
-				test1: 'i exist',
+			test('.keepFieldsContaining should work on a nested object', () => {
+				expect(
+					evaluate(
+						'={{ ({ test1: "i exist", test2: "i should be removed", test3: { test4: "me too" } }).keepFieldsContaining("exist") }}',
+					),
+				).toEqual({
+					test1: 'i exist',
+				});
 			});
-		});
 
-		test('.keepFieldsContaining should not work for empty string', () => {
-			expect(() =>
-				evaluate(
-					'={{ ({ test1: "i exist", test2: "i should be removed", test3: "i should also be removed" }).keepFieldsContaining("") }}',
-				),
-			).toThrow();
+			test('.keepFieldsContaining should not work for empty string', () => {
+				expect(() =>
+					evaluate(
+						'={{ ({ test1: "i exist", test2: "i should be removed", test3: "i should also be removed" }).keepFieldsContaining("") }}',
+					),
+				).toThrow();
+			});
 		});
 
-		test('.compact should work on an object', () => {
-			expect(
-				evaluate('={{ ({ test1: 1, test2: "2", test3: undefined, test4: null }).compact() }}'),
-			).toEqual({ test1: 1, test2: '2' });
+		describe('.compact', () => {
+			test('should work on an object', () => {
+				expect(
+					evaluate('={{ ({ test1: 1, test2: "2", test3: undefined, test4: null }).compact() }}'),
+				).toEqual({ test1: 1, test2: '2' });
+			});
+
+			test('should remove fields with null, undefined, empty string, or "nil"', () => {
+				expect(
+					evaluate(
+						'={{ ({ test1: 0, test2: false, test3: "", test4: "nil", test5: NaN }).compact() }}',
+					),
+				).toEqual({ test1: 0, test2: false, test5: NaN });
+			});
+
+			test('should work on an empty object', () => {
+				expect(evaluate('={{ ({}).compact() }}')).toEqual({});
+			});
+
+			test('should work on an object with all null/undefined values', () => {
+				expect(evaluate('={{ ({ test1: undefined, test2: null }).compact() }}')).toEqual({});
+			});
+
+			test('should work on an object with nested null/undefined values', () => {
+				expect(
+					evaluate(
+						'={{ ({ test1: 1, test2: { nested1: null, nested2: "value" }, test3: undefined }).compact() }}',
+					),
+				).toEqual({ test1: 1, test2: { nested2: 'value' } });
+			});
+
+			test('should not allow prototype pollution', () => {
```

Comment:
This is the main test that was added to assert that prototype pollution isn't possible via expressions.